File: 02 - api/python/api/climate/controller/climate_controller.py
# climate/controllers/climate_controller.py
from ninja import NinjaAPI

from climate.service.climate_service import ClimateService


api = NinjaAPI()
climate_service = ClimateService()

@api.get("/{country}")
def get_climate(request, country: str):
    return climate_service.get_climate(country)

@api.get("/rain/{id}")
def get_climate_rain(request, id: str):
    return climate_service.get_climate_rain(id)

File: 022024/02 - api/javaScrpit/climate/src/controller/climate_controller.js
import express from "express";
import ClimateService from "../service/climate_service.js";

const router = express.Router();
const climateService = new ClimateService();

router.get("/:country", async (req, res) => {
  const country = req.params.country;
  const data = await climateService.getClimate(country);
  res.json(data);
});

router.get("/rain/:id", async (req, res) => {
  const id = req.params.id;
  const data = await climateService.getClimateRain(id);
  res.json(data);
});

export default router;
